Unhandled exception. System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Program.Dangerous(Int32[] array, Int32 index) in /src/Program.cs:line 3
   at Program.Foo(Int32[] array, Int32 index) in /src/Program.cs:line 4
   at Program.Foo1(Int32[] array, Int32 index) in /src/Program.cs:line 5
   at Program.Foo2(Int32[] array, Int32 index) in /src/Program.cs:line 6
   at Program.Foo3(Int32[] array, Int32 index) in /src/Program.cs:line 7
   at Program.Foo4(Int32[] array, Int32 index) in /src/Program.cs:line 8
   at Program.Foo5(Int32[] array, Int32 index) in /src/Program.cs:line 9
   at Program.Foo6(Int32[] array, Int32 index) in /src/Program.cs:line 10
   at Program.Main() in /src/Program.cs:line 13
